concepts/Common/TypeScript Guides/10 Best Practices/05 Using Unions Instead of Base Classes.md

```diff
@@ -0,0 +1,5 @@
+Inheritance is part of internal implementation. You should not rely on the knowledge which component derives from which base. For generalization, use a union to define your own type.
```

Comment:
"knowledge which component derives from which base" sounds strange. I want to avoid "which", and there should be a preposition after "knowledge". 

```suggestion
Inheritance is a part of internal implementation. You should not rely on the knowledge of relationships between base classes and their descendants. For generalization, use a union to define your own type.
```